packages/adapters-library/src/adapters/metamask/products/validator-staking/metamaskValidatorStakingAdapter.ts

```diff
@@ -0,0 +1,146 @@
+import { getAddress } from 'ethers'
+import { AdaptersController } from '../../../../core/adaptersController'
+import { Chain } from '../../../../core/constants/chains'
+import { CacheToDb } from '../../../../core/decorators/cacheToDb'
+import { Helpers } from '../../../../core/helpers'
+import { CustomJsonRpcProvider } from '../../../../core/provider/CustomJsonRpcProvider'
+import {
+  IProtocolAdapter,
+  ProtocolToken,
+} from '../../../../types/IProtocolAdapter'
+import {
+  AdapterSettings,
+  GetPositionsInput,
+  PositionType,
+  ProtocolAdapterParams,
+  ProtocolDetails,
+  ProtocolPosition,
+  TokenType,
+  UnwrapExchangeRate,
+  UnwrapInput,
+} from '../../../../types/adapter'
+import { Protocol } from '../../../protocols'
+import { ZERO_ADDRESS } from '../../../../core/constants/ZERO_ADDRESS'
+
+export class MetamaskValidatorStakingAdapter implements IProtocolAdapter {
+  productId = 'validator-staking'
+  protocolId: Protocol
+  chainId: Chain
+  helpers: Helpers
+
+  adapterSettings: AdapterSettings = {
+    includeInUnwrap: true,
+    userEvent: {
+      topic0:
+        '0xac1020908b5f7134d59c1580838eba6fc42dd8c28bae65bf345676bba1913f8e',
+      userAddressIndex: 2,
+    },
+  }
+
+  private provider: CustomJsonRpcProvider
+
+  adaptersController: AdaptersController
+
+  constructor({
+    provider,
+    chainId,
+    protocolId,
+    adaptersController,
+    helpers,
+  }: ProtocolAdapterParams) {
+    this.provider = provider
+    this.chainId = chainId
+    this.protocolId = protocolId
+    this.adaptersController = adaptersController
+    this.helpers = helpers
+  }
+
+  getProtocolDetails(): ProtocolDetails {
+    return {
+      protocolId: this.protocolId,
+      name: 'Metamask',
+      description: 'Metamask defi adapter',
+      siteUrl: 'https://portfolio.metamask.io/',
+      iconUrl: 'https://portfolio.metamask.io/favicon.png',
+      positionType: PositionType.Staked,
+      chainId: this.chainId,
+      productId: this.productId,
+    }
+  }
+
+  @CacheToDb
+  async getProtocolTokens(): Promise<ProtocolToken[]> {
```

Comment:
```suggestion
  async getProtocolTokens(): Promise<[ProtocolToken]> {
```
Make this change and you won't have to do assertions in `getPositions`.